Source/LibationFileManager/PropertyChangeFilter.cs

```diff
@@ -0,0 +1,331 @@
+		void MyPropertiesChanged(object sender, PropertyChangedEventArgsEx e)
+		{
+			// Only properties whose names match either "MyProperty1"
+			// or "MyProperty2" will fire this event handler.
+		}
+
+	******
+	* OR *
+	******
+	
+		propertyChangeFilter.PropertyChanged +=
+			[PropertyChangeFilter("MyProperty1")]
+			[PropertyChangeFilter("MyProperty2")]
+			(_, _) => 
+			{
+				// Only properties whose names match either "MyProperty1"
+				// or "MyProperty2" will fire this event handler.
+			};
+
+
+		*************************
+		*						*
+		*    Observable Mode	*
+		*						*
+		*************************
+		
+		using var cancellation = propertyChangeFilter.ObservePropertyChanging<int>("MyProperty", MyPropertyChanging);
+		
+        void MyPropertyChanging(string propertyName, int oldValue, int newValue)
+        {
+			// Only the property whose name match
+			// "MyProperty" will fire this method.
+        }
+	
+		//The observer is delisted when cancellation is disposed
+
+	******
+	* OR *
+	******
+	
+		using var cancellation = propertyChangeFilter.ObservePropertyChanged<bool>("MyProperty", (_, s) =>
+			{
+				// Only the property whose name match
+				// "MyProperty" will fire this action.
+			});
+		
+		//The observer is delisted when cancellation is disposed
+
+	 */
+
+	#endregion
+
+	public abstract class PropertyChangeFilter
+	{
+		private readonly Dictionary<string, List<Delegate>> propertyChangedActions = new();
+		private readonly Dictionary<string, List<Delegate>> propertyChangingActions = new();
+		
+		private readonly List<KeyValuePair<PropertyChangedEventHandlerEx, PropertyChangedEventHandlerEx>> changedFilters = new();
+		private readonly List<KeyValuePair<PropertyChangingEventHandlerEx, PropertyChangingEventHandlerEx>> changingFilters = new();
+
+		public PropertyChangeFilter()
+		{
+			PropertyChanging += Configuration_PropertyChanging;
+			PropertyChanged += Configuration_PropertyChanged;
+		}
+
+		#region Events
+
+		protected void OnPropertyChanged(string propertyName, object newValue)
+			=> _propertyChanged?.Invoke(this, new(propertyName, newValue));
+		protected void OnPropertyChanging(string propertyName, object oldValue, object newValue)
+			=> _propertyChanging?.Invoke(this, new(propertyName, oldValue, newValue));
+
+		private PropertyChangedEventHandlerEx _propertyChanged;
+		private PropertyChangingEventHandlerEx _propertyChanging;
+
+		public event PropertyChangedEventHandlerEx PropertyChanged
+		{
+			add
+			{
+				var attributes = Attribute.GetCustomAttributes(value.Method, typeof(PropertyChangeFilterAttribute)) as PropertyChangeFilterAttribute[];
+
+				if (attributes.Any())
+				{
+					var matches = attributes.Select(a => a.PropertyName).ToArray();
+
+					void filterer(object s, PropertyChangedEventArgsEx e)
+					{
+						if (e.PropertyName.In(matches)) value(s, e);
+					}
+
+					changedFilters.Add(new(value, filterer));
+
+					_propertyChanged += filterer;
+				}
```

Comment:
If the handler has `PropertyChangeFilterAttribute`s, pass them through the `filterer` delegate and subscribe `filterer` to the event instead of `value`.